docs/developers/best_practices.md

```diff
@@ -45,19 +45,54 @@ sample for further guidance on using dynamic shapes.
 
 ## Practices for compilation settings
 
-TODO: mention parameters to tune
-
 TODO: which compiler targets to use (try both CUDA and Vulkan?)
 
 TODO: use the most specific LLVM target triple you can?
 
-## Practices for runtime use
+### Tuning compilation heuristics
```

Comment:
Sorry for the late reply!

As discussed in our 1:1, it would be nice to categorize stuff a bit:

1. Mention long term we want to support attaching attributes to the input model to allow autotuning. This should be the prefered way and there are work undergoing to make it happen.
2. CL options that supply a number. These are really knobs that can be tuned. We should mention preferably they should also be folded into the above category when possible. But there can exist outliers. So case by case.
3. CL options for experimental features. We should mention these should all be removed and making it as the default.